src/_pytest/nodes.py

```diff
@@ -417,6 +429,43 @@ def __init__(
 
         super().__init__(name, parent, config, session, nodeid=nodeid, fspath=fspath)
 
+        self._norecursepatterns = self.config.getini("norecursedirs")
+
+    def gethookproxy(self, fspath):
+        from _pytest.main import Session  # noqa: F811
+        from _pytest.python import Package
+
+        if not isinstance(self, (Session, Package)):
```

Comment:
From https://github.com/pytest-dev/pytest/pull/6461#discussion_r368104202 I understood that this would issue a deprecation unconditionally, regardless if self is `Session` or `Package`, no?

If we go with the deprecation then this should go back to `features`.

Reply:
@RonnyPfannschmidt said it should become not "more public", and said that it might get deprecated later.  I.e. this only cleans up the existing code.

Reply:
Having the extra imports there is rather ugly I think, but cannot be top-level.
Should this compare with `self.__class__` instead?

Reply:
I'm talking about this reply https://github.com/pytest-dev/pytest/pull/6461#discussion_r368104202:

> @blueyed good point about the deprecation warning, moving the method up while deprecating it is actually better, thanks for bringing it up

I would prefer we do it now anyway, as it is simpler (@RonnyPfannschmidt can correct me if I'm misunderstanding).

Reply:
I don't think we _need_ to deprecate it already, and IMHO it is also an example of unnecessarily breaking code.
As for why it should stay "less public" @RonnyPfannschmidt said:

> can we try to keep this as a private api, we need to cahnge the path type eventually to pathlib

So this cannot really be done now anyway, should it?

Reply:
> As for why it should stay "less public" @RonnyPfannschmidt said

That was posted before the comment I mentioned, so I assumed @RonnyPfannschmidt changed his mind.

Anyway, I'm removing myself from this review, I will let you guys decide on the outcome. 👍